File: server/src/main/java/moe/yuuta/server/formprocessor/FormData.java
package moe.yuuta.server.formprocessor;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface FormData {
    String value();
    boolean urlEncode() default false;
    boolean ignorable() default true;
}

File: server/src/main/kotlin/moe/yuuta/server/formprocessor/FormData.kt
package moe.yuuta.server.formprocessor


@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class FormData(val name: String,
                          val urlEncode: Boolean = false,
                          // If it is true, the fields with default values (String: "", Number: 0) will be removed.
                          // If it is false, the fields with default values will be kept as 'key=' or 'key=0', etc.
                          // Nulls are always removed.
                          val ignorable: Boolean = true)